src/collections.ts

```diff
@@ -307,7 +322,7 @@ export class Chain<T> {
 
   flatten(): FlattenChain<T> {
     const flattened = this.val.flatMap((it) =>
-      it instanceof Chain ? it.val : []
+      it instanceof Chain ? it.val : it
```

Comment:
We can't really be sure what type "it" is here because suse JavaScript, so we should check that it's actually an Array and use [] if it's not like before.